lib/common/client/errors.rb

```diff
@@ -10,22 +10,21 @@ class ClientError < Error
         attr_accessor :status
         attr_accessor :body
 
-        def initialize(message = nil, status = nil, body = nil)
+        def initialize(error = nil, message: nil, status: nil, body: nil)
           super(message)
+          @cause = error
```

Comment:
This class inherits from `StandardError`, so why do we have to *add* the cause attribute? 

It should be [built-in](https://ruby-doc.org/core-2.6/Exception.html) - add specs/documentation/comments to explain why this is necessary.